airbyte-webapp/src/views/Connector/ConnectorForm/useAuthentication.tsx

```diff
@@ -178,7 +182,7 @@ export const useAuthentication = (): AuthenticationHook => {
   const implicitAuthFieldPaths = useMemo(
     () => [
       // Fields from `advancedAuth` connectors
-      ...(advancedAuth
+      ...(advancedAuth && !isSourceDefinitionSpecificationDraft(connectorSpec)
```

Comment:
I didn't find a straightforward way to do so. This is an annoying property of hooks - they always need to be called in the same order, so you can't conditionally call them as you suggested without breaking the [rules of hooks](https://reactjs.org/docs/hooks-rules.html) which seems worse to me than putting this condition deeper into the tree.

I agree that it's annoying that it needs to be checked in a bunch of places, but Typescript is flagging all of them so we can't run into cases where code would compile fine, then crash when a draft specification is handed to it.

To avoid this, we could pull the check above the hook in a parent component, then not render the component if the definition is a draft. However, that would require us to refactor a bunch of components and we would need to pass the spec down as a prop because `useConnectorForm` can always return a draft.

Overall, while this solution isn't ideal it seemed like the best approach without bigger changes in the structure.

Reply:
Got it, your explanation makes sense. It's a good callout that typescript will flag any place that we forgot to add a check for this, so that will at least ensure we are consistent.

I don't think doing a big refactor of many components is worth it just to remove some of these conditionals, so I'm in agreement with you here 👍